web/api/v1/api.go

```diff
@@ -108,6 +108,31 @@ type rulesRetriever interface {
 	AlertingRules() []*rules.AlertingRule
 }
 
+// PrometheusVersion contains build information about Prometheus.
+type PrometheusVersion struct {
+	Version   string `json:"version"`
+	Revision  string `json:"revision"`
+	Branch    string `json:"branch"`
+	BuildUser string `json:"buildUser"`
+	BuildDate string `json:"buildDate"`
+	GoVersion string `json:"goVersion"`
+}
+
+type RuntimeInfo struct {
```

Comment:
@brian-brazil Thank you for your feedback! Can you please share your thoughts about using string Map instead of struct. Generally, I took it from the current status Handler (responsible for executing status.html template in web.go) and use it as a type of what API should receive.

Reply:
I don't see how it would work well for `RuntimeInfo` to be a map of strings (at least for the values), given that some values are not strings, and should be represented as non-strings in JSON. For `PrometheusVersion` that might work better, but I think a Go struct is fine there too.

I also share the concern of "this is a big bag of values, you can't rely on these staying the same over Prometheus versions". Not sure how to signal that besides documenting it clearly (and/or moving this endpoint outside of `/api/v1`, but I'd rather keep it as part of the v1 API I think).

Reply:
This belongs in v1 one way or the other.

Reply:
Yeah, would think so as well. So we'll just document that the fields can change with a big warning then.